monai/networks/blocks/dints_block.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import torch.nn as nn
+
+__all__ = ["FactorizedIncreaseBlock", "FactorizedReduceBlock", "P3DReLUConvBNBlock", "ReLUConvBNBlock"]
+
+
+class FactorizedIncreaseBlock(nn.Module):
+    """
+    Up-sampling the feature by 2 using stride.
+    """
+
+    def __init__(self, in_channel: int, out_channel: int):
+        super().__init__()
+        self._in_channel = in_channel
+        self._out_channel = out_channel
+        self.op = nn.Sequential(
+            nn.Upsample(scale_factor=2, mode="trilinear", align_corners=True),
+            nn.ReLU(),
+            nn.Conv3d(self._in_channel, self._out_channel, 1, stride=1, padding=0, bias=False),
+            nn.InstanceNorm3d(self._out_channel),
+        )
+
+    def forward(self, x):
+        return self.op(x)
+
+
+class FactorizedReduceBlock(nn.Module):
+    """
+    Down-sampling the feature by 2 using stride.
+    """
+
+    def __init__(self, in_channel: int, out_channel: int):
+        super().__init__()
+        self._in_channel = in_channel
+        self._out_channel = out_channel
+
+        assert self._out_channel % 2 == 0
+        self.relu = nn.ReLU()
+        self.conv_1 = nn.Conv3d(self._in_channel, self._out_channel // 2, 1, stride=2, padding=0, bias=False)
+        self.conv_2 = nn.Conv3d(self._in_channel, self._out_channel // 2, 1, stride=2, padding=0, bias=False)
+        self.bn = nn.InstanceNorm3d(self._out_channel)
```

Comment:
the activation function and feature norm function are currently hard-coded, is this a design choice or should this be flexible, with options such as https://github.com/Project-MONAI/MONAI/blob/33b6d61fdd8a15247e63bf7db16ac32f98f83566/monai/networks/blocks/dynunet_block.py#L48-L49